pkg/build/controller/build_controller.go

```diff
@@ -130,6 +132,16 @@ func (bc *BuildController) HandleBuild(build *buildapi.Build) error {
 		return err
 	}
 
+	// Handle processing ValueFrom in build strategy environment variables
+	originalEnvVars := buildutil.GetEnvFromStrategy(build)
+	processedEnvVars, err := common.ProcessValueFrom(build, bc.KubeClient)
+	if err != nil {
+		build.Status.Phase = buildapi.BuildPhaseFailed
+		build.Status.Reason = buildapi.StatusReasonInvalidEnvironmentVariable
+		return err
+	}
+	*originalEnvVars = *processedEnvVars
```

Comment:
The code for this is getting refactored with my pull https://github.com/openshift/origin/pull/14289. At some point we'll need to reconcile.